### ISSUE TITLE
[unbound-method] false positive when overwriting a bound method

### ISSUE THREAD

**Issue by dinofx** [typescript-eslint/typescript-eslint#1894]
Referencing and assigning a method are two different things.  Reading `stdout.write` is flagged, but overwriting it to some monkey-patched function should not be.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": ["error"]
  }
}
```

```TS
  const { stdin, stdout } = process;
  const stdoutWrite = stdout.write.bind(stdout);
  //...

// assignment below marked as error
  stdout.write = function(...args: any) {
    // Do something cool first
    stdoutWrite.apply(null, args);
    // ...
  } as typeof stdoutWrite;
```

**Expected Result**

Overwriting the originally-bound function should not be flagged as referencing the bound function.

**Actual Result**

false positive

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.9.0` |
| `TypeScript`                       | `3.8.3` |

**Comment by bradzacher** [comment]
could you please try updating to the latest version? I believe this is fixed in `v2.28.0`
